src/decoders/belief_propagation.jl

```diff
@@ -108,6 +108,71 @@ true
 function decode!(decoder::BeliefPropagationDecoder, syndrome::AbstractVector, error::AbstractVector) # TODO check if casting to bitarrays helps with performance -- if it does, set up warnings to the user for cases where they have not done the casting
   reset!(decoder)
   success = syndrome_decode!(decoder, decoder.scratch, syndrome) # TODO: Delete syndrome_decode! and just move its content here. There is no point in this indirection.
```

Comment:
I am confused by this line. Did you forget to delete it?

Reply:
also, if that is the last place where `syndrome_decode!` is being used, it would be nice to also delete that function and its dependencies altogether

Reply:
Removed `syndrome_decode!` line. Although not deleting the method as it is being used in the batchdecode format.